src/components/primitives/lists/AccordionListItem.tsx

```diff
@@ -0,0 +1,125 @@
+/*
+ * Copyright (c) Meta Platforms, Inc. and affiliates.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+import * as React from 'react';
+import stylex from '@stylexjs/stylex';
+import {fontStyles} from '../styles';
+import {textColors} from '../colors.stylex';
+import {AccordionListContext} from './AccordionListContext';
+
+export interface AccordionListItemProps {
+  /**
+   * Unique identifier for the list item.
+   */
+  id: string;
+  /**
+   * The label text to display for the list item.
+   */
+  label: string;
+  /**
+   * Optional sublabel text to display for the list item.
+   */
+  sublabel?: string;
+  /**
+   * Optional icon to display at the start of the list item.
+   */
+  startIcon?: React.ReactElement;
+  /**
+   * Optional badge to display in the list item.
+   */
+  badge?: React.ReactElement;
+  /**
+   * Optional icon to display at the end of the list item.
+   */
+  endIcon?: React.ReactElement;
+  /**
+   * The children items to render. Can be a single item or an array of items.
+   */
+  children: React.ReactNode | React.ReactNode[];
+}
+
+export default function AccordionListItem({
+  id,
+  label,
+  sublabel,
+  startIcon,
+  badge,
+  endIcon,
+  children,
+}: AccordionListItemProps) {
+  const {expandedItemId, onExpandedItemChange} =
+    React.useContext(AccordionListContext);
+
+  return (
+    <>
+      <div
+        {...stylex.props(styles.item)}
+        onClick={() => onExpandedItemChange?.(id)}
+      >
+        {startIcon && <>{startIcon}</>}
+        <div {...stylex.props(styles.center)}>
+          <span {...stylex.props(fontStyles.body, styles.label)}>{label}</span>
+          {sublabel && (
+            <span {...stylex.props(fontStyles.supporting, styles.sublabel)}>
+              {sublabel}
+            </span>
+          )}
+        </div>
+        {badge && <>{badge}</>}
+        {endIcon && <>{endIcon}</>}
```

Comment:
The truthiness check and fragments aren't really necessary here, either way it's undefined or a ReactElement.